### DISCUSSION TITLE
Can I know when all reactions for the current transaction have run?

### DISCUSSION THREAD

**Discussion by rickmed** [mobxjs/mobx#3508] [Q&A] [answered]
Like a hook or notification?

**Comment by urugator** [comment, marked as answer]
https://mobx.js.org/configuration.html#reactionscheduler-f---void--void
```javascript
configure({
    reactionScheduler: (runReactions): void => {
        console.log("before");
        runReactions();
        console.log("after");
    }
})
```

**Reply by rickmed** [reply]
I could also wrap all functions that wrapped in actions so that when they're are finished I can run some code right? (ie, actions are synchronous)

**Reply by urugator** [reply]
Yes, but keep in mind that only the outermost action counts, so you either have to keep this wrapper at the "boundary" or it should also support nesting. Just in case, you can get the action's nesting counter as `mobx._getGlobalState().inBatch`
https://github.com/mobxjs/mobx/blob/f0e066f427d573cbef4b92f3310eb069c3aca205/packages/mobx/src/core/globalstate.ts#L64
